test/functional/tests/trim/test_trim.py

```diff
@@ -0,0 +1,105 @@
+#
+# Copyright(c) 2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import pytest
+import re
+
+from api.cas import casadm
+from core.test_run import TestRun
+from test_utils import os_utils
+from test_utils.size import Size, Unit
+from test_tools.fio.fio import Fio
+from test_tools.blktrace import BlkTrace, BlkTraceMask
+from test_tools.fio.fio_param import ReadWrite, IoEngine
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+
+
+#TODO Add mark for getting discard-supproting devcie
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+def test_trim_start_discard():
+    """
+    title: Check discarding cache device at cache start
+    description: |
+       Create 2 partitions on trim-supporting device, write pattern to both partitions,
+       start blktrace against first one, start cache on first partition and check if discard
+       requests were sent at all and only to the first partition.
+    pass_criteria:
+      - Partition used for cache is discarded.
+      - Second partition is untouched - written pattern is preserved.
+    """
+    with TestRun.step("Clearing dmesg"):
+        TestRun.executor.run_expect_success("dmesg -C")
+
+    with TestRun.step("Preparing cache device"):
+        dev = TestRun.disks['cache']
+        dev.create_partitions([Size(500, Unit.MebiByte), Size(500, Unit.MebiByte)])
+        cas_part = dev.partitions[0]
+        non_cas_part = dev.partitions[1]
+
+    with TestRun.step("Writing different pattern on partitions"):
+        cas_fio = write_pattern(cas_part.system_path)
+        non_cas_fio = write_pattern(non_cas_part.system_path)
+        cas_fio.run()
+        non_cas_fio.run()
+
+    # TODO add blktracing for non-cas part
+    with TestRun.step("Starting blktrace against first (cache) partition"):
+        blktrace = BlkTrace(cas_part, BlkTraceMask.discard)
+        blktrace.start_monitoring()
+
+    with TestRun.step("Starting cache"):
+        cache = casadm.start_cache(cas_part, force=True)
+        metadata_size = get_metadata_size_from_dmesg()
+
+    with TestRun.step("Stop blktrace and check if discard requests were issued"):
+        cache_reqs = blktrace.stop_monitoring()
+        cache_part_start = cas_part.begin
+
+        if len(cache_reqs) == 0:
+            TestRun.fail("No discard requests issued to the cas partition!")
+
+        non_meta_sector = (cache_part_start + metadata_size).get_value(Unit.Blocks512)
+        non_meta_size = cas_part.size - metadata_size
```

Comment:
You are using `.get_value(Unit.Byte)` later anyway, why don't You call it here? It would be more consistent